srv/components/provisioner/scripts/cortx_prep.sh

```diff
@@ -0,0 +1,306 @@
+                echo "ERROR: cortx_setup command still not available " | tee -a ${LOG_FILE}
+                echo "ERROR: Please check if cortx_setup commands"\
+                " are installed at correct location and environment variable PATH is updated." | tee -a ${LOG_FILE}
+                exit 1
+            fi
+    fi
+    echo "Done" | tee -a ${LOG_FILE}
+}
+
+download_isos()
+{
+    CORTX_RELEASE_REPO=$1
+    echo "Downloading the ISOs" | tee -a ${LOG_FILE}
+    mkdir /opt/isos
+    cortx_iso=\$(curl -s ${CORTX_RELEASE_REPO}/iso/ | sed 's\/<\\/*[^>]*>\/\/g' | cut -f1 -d' ' | grep 'single.iso')
+    curl -O ${CORTX_RELEASE_REPO}/iso/\${cortx_iso}
+    os_iso=\$(curl -s ${CORTX_RELEASE_REPO}/iso/ | sed 's\/<\\/*[^>]*>\/\/g' | cut -f1 -d' '|grep  "cortx-os")
+    curl -O ${CORTX_RELEASE_REPO}/iso/\${os_iso}
+    #CORTX_PREP=\$(curl -s ${CORTX_RELEASE_REPO}/iso/ | sed 's\/<\\/*[^>]*>\/\/g' | cut -f1 -d' '|grep  ".sh")
+    #curl -O ${CORTX_RELEASE_REPO}/iso/\${CORTX_PREP}
+}
+
+main()
+{
+    echo "Running Cortx-prep script" | tee -a ${LOG_FILE}
+    parse_args $@
+    #TODO: uncomment once ready
+    #    create_factory_user $user_name $uid $group $gid
+    if hostnamectl status | grep Chassis | grep -q server; then
+        echo "DEBUG: This is Hardware" >> ${LOG_FILE}
+        if systemctl list-units | grep scsi-network-relay; then
+            systemctl start scsi-network-relay
+        else
+            echo "ERROR scsi-network-relay service is not present" | tee -a ${LOG_FILE}
+            exit 1
+        fi
+        if [[ ! -z $repo_url ]]; then
+            # User has provided target-build for HW
+            # Remove the ISOs set up by Kickstart
+            # download the iso from provided build url
+            # mount the iso
+            
+            #echo "Removing the ISOs setup by Kickstart" | tee -a ${LOG_FILE}
+            #TODO:
+                #remove_isos
+                #download_isos $repo_url
+                #mount_isos
+                #configure repos from iso files
+
+            #WORKAROUND: Use hosted repos until download_isos function is ready
+            setup_repos $repo_url
+        else
+            echo "DEBUG: Using the ISO files set up by kickstart " >> ${LOG_FILE}
+            #TODO: use the same isos downloaded by KS file to install the packages
+            # configure repos from iso files
+
+            echo "ERROR: Please provide the target-build unless kickstart file is in place" | tee -a ${LOG_FILE}
+            exit 1
+        fi
+    else
+        echo "This is VM" >> ${LOG_FILE}
+        if [[ ! -z $repo_url ]]; then
+            setup_repos $repo_url
+        else
+            echo "ERROR: target-build not provided, it's mandetary option for VM"
+            echo "ERROR: Please provide the target-build " | tee -a ${LOG_FILE}
+            exit 1
+        fi
+    fi
+
+    install_dependency_pkgs
+    install_cortx_pkgs
+    config_local_salt
+
+
```

Comment:
Just after configuring local salt please and before prepare **ConfStore** execute this command:
`components.provisioner.config.machine_id.reset`